app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,522 @@
+                .observeOn(AndroidSchedulers.mainThread())
+                .flatMap { response ->
+                    if (!response.continuation().isNullOrEmpty()) {
+                        articleContributionsContinuation = response.continuation()!!["uccontinue"]
+                    }
+                    continuedArticlesContributions.clear()
+                    for (userContribution in response.query()!!.userContributions()) {
+                        var descLang = ""
+                        val strArr = userContribution.comment.split(" ")
+                        for (str in strArr) {
+                            if (str.contains("wbsetdescription")) {
+                                val descArr = str.split("|")
+                                if (descArr.size > 1) {
+                                    descLang = descArr[1]
+                                    break
+                                }
+                            }
+                        }
+                        if (descLang.isEmpty()) {
+                            continue
+                        }
+
+                        if (!qLangMap.containsKey(userContribution.title)) {
+                            qLangMap[userContribution.title] = HashSet()
+                        }
+                        continuedArticlesContributions.add(Contribution(userContribution.title, "", "", EDIT_TYPE_ARTICLE_DESCRIPTION,
+                                "", DateUtil.iso8601DateParse(userContribution.timestamp), WikiSite.forLanguageCode(descLang), 0, 0))
+
+                        qLangMap[userContribution.title]!!.add(descLang)
+                    }
+                    ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).getWikidataLabelsAndDescriptions(qLangMap.keys.joinToString("|"))
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                }.subscribe({
+                    for (entityKey in it.entities().keys) {
+                        val entity = it.entities()[entityKey]!!
+                        for (contribution in continuedArticlesContributions) {
+                            if (contribution.qNumber == entityKey) {
+                                contribution.title = entity.labels()[contribution.wikiSite.languageCode()]!!.value()
+                                contribution.description = entity.descriptions()[contribution.wikiSite.languageCode()]!!.value()
+                            }
+                        }
+                    }
+                    getArticleContributionDetails()
+                }, { t ->
+                    L.e(t)
+                    showError(t)
+                }))
+    }
+
+    private fun getArticleContributionDetails() {
+        var count = 0
+        for (contributionObject in continuedArticlesContributions) {
+            disposables.add(ServiceFactory.getRest(contributionObject.wikiSite).getSummary(null, contributionObject.title)
```

Comment:
Would it be possible to use `Observable.fromIterable(continuedArticlesContributions).flatMap{...}` for this instead of using for loop?